Implement macOS grep polyfill and enhance debug output [BUGFIX]

bdteo opened the pull request:
This pull request addresses an issue where the PHP version was incorrectly determined on macOS within the configure_package function. The root cause was that macOS's native grep command lacks support for the -P option.
To resolve this, the following changes have been made:

Implemented a polyfill for macOS's version of grep to ensure compatibility. The polyfill modifies the grep arguments by replacing -P with -E when necessary.
Enhanced the debug output by adding more detailed information during the debugging process. The added debug output provides valuable insights into variables and their values, facilitating easier troubleshooting.
Updated the .gitignore file to exclude JetBrains IDE and Mac OS X system files, keeping the repository clean and focused on relevant files.

These changes improve the reliability and usability of the php-build tool on macOS systems. The polyfill ensures that the PHP version is correctly determined, while the enhanced debug output aids in identifying and resolving issues more efficiently.
Please review and consider merging these changes to enhance the compatibility and debugging capabilities of php-build on macOS.

Best Regards,
@bdteo

bdteo commented:
@GrahamCampbell, can you please take a look when you have the opportunity.
All the best! Props for your work on the project!

GrahamCampbell commented on .gitignore:
```diff
@@ -13,3 +13,10 @@ share/php-build/before-install.d/*
 share/php-build/after-install.d/*
 !share/php-build/after-install.d/.empty
+
+# JetBrains IDE
```
please revert the changes to the gitignore

bdteo commented:
Hi @GrahamCampbell,

I've addressed the macOS compatibility issue with the `grep` command in the `configure_package` function by implementing a polyfill that replaces `-P` with `-E`. The changes resolve the incorrect PHP version determination caused by the lack of `-P` support in macOS's `grep`.

I've also reverted the `.gitignore` to its prior state as per your request.

Please review the pull request and provide any feedback or suggestions. A few points to consider:

1. Error handling for `command grep` failures.
2. Efficiency of directly calling `grep` with modified arguments.
3. Structured logging or configuration flag for debug output.
4. Consistency with existing codebase's coding style and formatting.
5. Brief description of the issue and solution in the pull request or documentation.

This macOS compatibility fix is crucial for improving the user experience. Your thoughts on addressing any of the suggestions would be appreciated.

Thank you for your time and consideration.

Best regards,
Boris D. Teoharov

bdteo commented:
Hi @GrahamCampbell,

Just to provide some context:

I reverted the `.gitignore` changes using rebase and force-push to prevent potential conflicts with other contributors' changes, as the fork is solely owned by me.

Commit 934e0096975602969d5b16e2780d5153bec86215 is a merge of the upstream master into my fork to ensure it's up to date before merging.

Let me know if you have any further questions.

Best regards,
Boris D. Teoharov

madumlao commented:
@bdteo i had a quick look at php-build usage of grep and it never uses the advanced perl regex functionality (`-P`) anyways. Is there anything wrong with straightup replacing -P with -E, which is more comptable?

I used this in my PR#790

madumlao commented:
Hi @bdteo , I have gone with the simpler, more compatible approach of using grep -E instead.
